apps/joint_states_subscriber/app.c

```diff
@@ -0,0 +1,118 @@
+#include <rcl/rcl.h>
+#include <rcl/error_handling.h>
+#include <rclc/rclc.h>
+#include <rclc/executor.h>
+
+#include <std_msgs/msg/string.h>
+#include <sensor_msgs/msg/joint_state.h>
+#include <stdio.h>
+
+#define ARRAY_LEN 200
+#define JOINT_DOUBLE_LEN 20
+
+#define RCCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Aborting.\n",__LINE__,(int)temp_rc); return 1;}}
+#define RCSOFTCHECK(fn) { rcl_ret_t temp_rc = fn; if((temp_rc != RCL_RET_OK)){printf("Failed status on line %d: %d. Continuing.\n",__LINE__,(int)temp_rc);}}
+
+rcl_subscription_t subscriber;
+std_msgs__msg__String msg;
+
+rcl_subscription_t joint_states_subscriber;
+sensor_msgs__msg__JointState joint_states_msg;
+
+char test_array[ARRAY_LEN];
+
+void subscription_callback(const void * msgin)
+{
+	const std_msgs__msg__String * msg = (const std_msgs__msg__String *)msgin;
+	printf("I have heard: \"%s\"\n", msg->data.data);
+}
```

Comment:
Maybe the example is more interesting if we only subscribe to JointStates in this example. Could you refactor?